x/circuit/keeper/msg_server.go

```diff
@@ -70,40 +75,52 @@ func (srv msgServer) AuthorizeCircuitBreaker(goCtx context.Context, msg *types.M
 	}, nil
 }
 
-func (srv msgServer) TripCircuitBreaker(goCtx context.Context, msg *types.MsgTripCircuitBreaker) (*types.MsgTripCircuitBreakerResponse, error) {
-	ctx := sdk.UnwrapSDKContext(goCtx)
-
+func (srv msgServer) TripCircuitBreaker(ctx context.Context, msg *types.MsgTripCircuitBreaker) (*types.MsgTripCircuitBreakerResponse, error) {
 	address, err := srv.addressCodec.StringToBytes(msg.Authority)
 	if err != nil {
 		return nil, err
 	}
 
 	// Check that the account has the permissions
-	perms, err := srv.GetPermissions(ctx, address)
-	if err != nil {
-		return nil, fmt.Errorf("user permission does not exist %w", err)
+	perms, err := srv.Permissions.Get(ctx, address)
+	if err != nil && !errorsmod.IsOf(err, collections.ErrNotFound) {
+		return nil, err
 	}
```

Comment:
Before it would return an error if the permissions wasn't found but now it does not. I think it still should.

Reply:
So if we return here we can't check later if the account that tripped the breaker is the Authority account.

It was a mistake that it returns "user permission does not exist" on error, because when no permissions were found it was returning an empty permission struct with a nil error.

See the original GetPermissions:

```go
func (k *Keeper) GetPermissions(ctx sdk.Context, address []byte) (*types.Permissions, error) {
	store := ctx.KVStore(k.storekey)

	key := types.CreateAddressPrefix(address)
	bz := store.Get(key)

	perms := &types.Permissions{}
	if err := proto.Unmarshal(bz, perms); err != nil {
		return &types.Permissions{}, err
	}

	return perms, nil
}
```